File: src/main/java/org/usfirst/frc/team3863/commands/TransmissionCommand.java
package org.usfirst.frc.team3863.commands;

import edu.wpi.first.wpilibj.DoubleSolenoid;
import org.usfirst.frc.team3863.Utils;

/**
 * Originally Created by Joshua Freedman on 2/9/2016.
 * Project: 2016Robot
 */
public class TransmissionCommand extends BaseCommand {

    private boolean enable = false;

    public TransmissionCommand(boolean enable) {
        requires(driveTrain);
        this.enable = enable;
    }

    @Override
    protected void initialize() {

    }

    @Override
    protected void execute() {
        System.err.println("WORKING");
        if (enable) {
            Utils.fireSolenoid(driveTrain.getDriveTrainSolenoid(), DoubleSolenoid.Value.kForward);
        } else {
            Utils.fireSolenoid(driveTrain.getDriveTrainSolenoid(), DoubleSolenoid.Value.kReverse);
        }
    }

    @Override
    protected boolean isFinished() {
        return true;
    }

    @Override
    protected void end() {

    }

    @Override
    protected void interrupted() {

    }
}

File: src/main/kotlin/org/usfirst/frc/team3863/kotlin/commands/TransmissionCommand.kt
package org.usfirst.frc.team3863.kotlin.commands

import edu.wpi.first.wpilibj.DoubleSolenoid

/**
 * Created by Joshua Freedman on 2/11/2016.
 * Project: 2016Robot
 */
class TransmissionCommand(val enable: Boolean) : BaseCommand("Transmission Command") {

    init {
        requires(pneumatics)
    }

    override fun initialize() {

    }

    override fun execute() {
        if (enable) {
            pneumatics.fireSolenoid(pneumatics.driveTrainSolenoid, DoubleSolenoid.Value.kForward)
        } else {
            pneumatics.fireSolenoid(pneumatics.driveTrainSolenoid, DoubleSolenoid.Value.kReverse)
        }
    }

    override fun isFinished(): Boolean {
        return true
    }

    override fun end() {

    }

    override fun interrupted() {

    }
}
